lms/djangoapps/mobile_api/users/urls.py

```diff
@@ -6,7 +6,7 @@
 from django.conf import settings
 from django.urls import re_path
 
-from .views import UserCourseEnrollmentsList, UserCourseStatus, UserDetail
```

Comment:
@KyryloKireiev If I understand right, the only new information that the UserEnrollmentsStatus API provides is `is_active`, with this logic:

> Returns active enrolment status if user was enrolled for the course
> less than 30 days ago or has progressed in the course in the last 30 days.
> Otherwise, the registration is considered inactive.

Rather than adding a new API endpoint, could this new field be added to the existing `UserCourseEnrollmentsList` API instead?

Reply:
@kdmccormick yes, you are right, the only new information is `is_active` status. But our architects decided we need to create a new API. A mobile team needs an API with the following properties:
1. The API should be fast and easy;
2. The API should return all users enrolments without pagination;
3. It would be nice if the API did not return anything unnecessary except `course name`, `course id` and `enrollment status`.

So, we decided to create a simple interface with only one responsibility.

`UserCourseEnrollmentsList` interface has already become too slow, heavy and universal. Also, pagination has already appeared in versions 3 and 4. That is, to obtain the necessary information, we would have to make an additional request to an older version of this API (v0.5, v1 or v2). It would also be necessary to add the “is_active” field to some of the older versions of this API(v0.5, v1 or v2), which would also violate the Open closed Principle.

Reply:
Hi @kdmccormick! Please give us an answer if possible. What do you think about our architectural solution?

Reply:
@KyryloKireiev The app is calling UserCourseEnrollmentsList anyway in order to render the course cards, right? Why do you need a separate un-paginated endpoint for the status?

Perhaps I am missing the bigger picture. Do you have a doc that outlines all the new and existing API endpoints you need?

Reply:
Hi @kdmccormick, thank you for your effort to finalize the reviews

So sorry that the context for FC-0047 is not completely clear. I'll try to address all questions and concerns.

As mentioned by Kyrylo above, this PR adds a new endpoint due to performance concerns with using UserEnrollmentsStatus API endpoint, that was developed and used to display User courses on mobile dashboard. Furthermore UserEnrollmentsStatus API [was extended with Primary course entry in previous FC-0047 PR](https://github.com/openedx/edx-platform/pull/34848/files#diff-e38b66be10e228f1728861821289555db4f2741b9057e1ecc70257d8ab52ddc8R264) to cover product requirements for Mobile Dashboard.


To find more details on this specific functionality which is supported by this new API endpoint, please refer to the video and screenshots in the description for mobile PR https://github.com/openedx/openedx-app-ios/pull/466.
Basically, this endpoint is used only inside the calendar synchronization view, providing the list of active courses that can be selected by a student to synchronize dates for these courses to Apple / Google calendars through native mobile functionality.


For now we only have a document with new PRs for FC-0047 that shows relation of the edx-platform API PR to Mobile APPs PRs in which user facing functionality is added  https://docs.google.com/spreadsheets/d/1ImoFKqZZnP3MDnPe_kUmmmuZBgV2teDqsOJVF8y6NjI/edit?gid=0#gid=0

I'll try to outline a document with all the API endpoints that were used or extended as part of FC-0047 today or tomorrow.

Reply:
@GlugovGrGlib 

> I'll try to outline a document with all the API endpoints that were used or extended as part of FC-0047 today or tomorrow.

That would be really helpful, thanks Glib. Can you make that document somewhere that permits review, like an adr, the wiki, or a google doc? I would like to review the API design as a whole, and once we are aligned there, I should be able to approve the individual PRs much more quickly.

Reply:
@GlugovGrGlib were you able to make a doc outlining the FC-0047 mobile API endpoints?

Reply:
@KyryloKireiev Thanks for that background. 

### Pagination/limits

In order to keep large Open edX sites safe from DDOS attacks, all HTTP APIs need to be paginated (with a max page size) or limited (for example, only return the 1000 most relevant results).

Without pagination or limiting, a bad-faith actor (or a sufficiently advanced good-faith user) could take down the site by creating a huge number of enrollments and then repeatedly calling the unbounded API.


### user_enrollments API versus enrollments_status

With this PR merged, we would have two separate APIs for listing enrollments: user_enrollments and enrollments_status. I want to understand whether we truly need both APIs.

Once enrollments_status is used by the new apps, will they still use user_enrollments too?
* If no, can user_enrollments be deprecated?
* If yes, then is user_enrollments always called alongside enrollments_status?
  *  If yes, then why does the speed of user_enrollments matter, since you will always need both API responses anyway?